File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        answer = []
        for num in nums:
            if num == val:
                continue
            answer.append(num)

        for i in range(len(answer)):
            nums[i] = answer[i]

        return len(answer)

File: Data Structures & Algorithms/majority-element-ii/submission-3.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> res;
        int totalCount = nums.size();
        unordered_map<int,int> counter;

        for (int num : nums) {
            counter[num] += 1;
        }

        for (auto [num,freq] : counter ) {
            double percentage = (double)freq / totalCount;
            if (percentage > 1.0 / 3.0) {
                res.push_back(num);
            }
        }

        return res;
    }
};